File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.java
class Solution {
    public int solution(int a, int b, int c) {
        int answer = 0;
        
        if(a != b && a != c && b != c) {
            answer = a + b + c;
        } else if(a == b && b == c) {
            answer = (a + b + c) * powNum(a, b, c, 2) * powNum(a, b, c, 3);
        } else {
            answer = (a + b + c) * powNum(a, b, c, 2);
        }
        
        return answer;
    }
    
    public int powNum(int a, int b, int c, int power) {
        return (int)(Math.pow(a , power) + Math.pow(b , power) + Math.pow(c , power));
    }
}

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.py
def solution(a, b, c):
    answer = a + b + c
    
    if a == b == c:
        answer = (a + b + c) * (a**2 + b**2 + c**2) * (a**3 + b**3 + c**3)
    else:
        answer = (a + b + c) * (a**2 + b**2 + c**2)
    
    return answer

# 다른 풀이
check=len(set([a,b,c]))
if check==1:
    return 3*a*3*(a**2)*3*(a**3)
elif check==2:
    return (a+b+c)*(a**2+b**2+c**2)
else:
    return (a+b+c)
